py4j-java/src/main/java/py4j/CallbackConnection.java

```diff
@@ -95,8 +100,9 @@ public CallbackConnection(int port, InetAddress address, SocketFactory socketFac
 	 *            program is closed. If readTimeout = 0, a default readTimeout of 1000 is used for operations that
 	 *            must absolutely be non-blocking.
 	 */
-	public CallbackConnection(int port, InetAddress address, SocketFactory socketFactory, int readTimeout) {
-		this(port, address, socketFactory, readTimeout, null);
+	public CallbackConnection(int port, InetAddress address, SocketFactory socketFactory, int readTimeout,
+			int createSocketTimeout) {
```

Comment:
likely also need to avoid breaking API here to keep back compat & pass default connect timeout.